### PR TITLE
FIX: fix passing empty lists to errorbar xerr/yerr

### LINKED ISSUE #5641: plt.errorbar error with empty list
The command below returns an error in Matplotlib 1.5.0
```
plt.errorbar([], [], xerr=[], yerr=[])
```
Even tough 
```
plt.errorbar([], [])
```
works as expected.

When the x and y coordinates are empty lists, It would make sense for the command to accept empty ```xerr``` and ```yerr``` keywords as well, without throwing an error, as in the previous Matplotlib version. In fact the empty lists will likely come from code like the following, where ```np.any(mask)==False```
```
mask = y > 0
plt.errorbar(x[mask], y[mask], xerr=xerr[mask], yerr=yerr[mask])
```

### PR DESCRIPTION
**Pull request by tacaswell**
closes #5641

### CODE REVIEW

**Comment by mdboom** [comment]
Looks fine to me as a stopgap until we have a more general solution, once tests pass.

**Comment by mdboom** [comment]
Restarted seemingly unrelated failure.

File: lib/matplotlib/axes/_axes.py [hunk 1]
```diff
@@ -2889,9 +2889,11 @@ def xywhere(xs, ys, mask):
                 # Check if xerr is scalar or symmetric. Asymmetric is handled
                 # above. This prevents Nx2 arrays from accidentally
                 # being accepted, when the user meant the 2xN transpose.
-                if not (len(xerr) == 1 or
-                        (len(xerr) == len(x) and not (
-                            iterable(xerr[0]) and len(xerr[0]) > 1))):
+                # special case for empty lists
+                if len(xerr) and not (len(xerr) == 1 or
```
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py]: If I'm not mistaken, these two conditions can be combined into `if len(xerr) > 1 and not (everything on the next two lines):`.

Comment (tacaswell) [inline lib/matplotlib/axes/_axes.py]: I think you are right.

**Comment by tacaswell** [comment]
There is some pathology I don't understand

```python
ax.errorbar(np.ones(5), np.ones(5), xerr=[[1]]*5)  # works
ax.errorbar([1]*5, [1]*5, xerr=[[1]]*5)  # fails
```

but I think we should let that go for now

**Pull request merged**